testing/integration/package_version_test.go

```diff
@@ -192,6 +192,25 @@ func TestComponentBuildHashInDiagnostics(t *testing.T) {
 		assert.Equalf(t, wantBuildHash, c.State.VersionInfo.Meta.Commit,
 			"component %s: VersionInfo.Meta.Commit mismatch", c.ID)
 	}
+	if t.Failed() {
+		diagPath, err := f.DiagnosticsDir()
+
+		err = os.MkdirAll(diagPath, 0755)
+		require.NoError(t, err, "could not keep the diagnostics used in the test so it'd be available after the test")
```

Comment:
Perhaps a more fitting failure message would be "failed to create diagnostics directory to preserve test artifacts" or something along these lines, wdyt?